src/components/CustomerQuota/NoQuota/NoQuotaCard.test.ts

```diff
@@ -2,62 +2,258 @@ import {
   defaultIdentifier,
   defaultTranslationProps,
 } from "../../../test/helpers/defaults";
-import { PastTransactionsResult, CampaignPolicy } from "../../../types";
+import { PastTransactionsResult, CampaignPolicy, Quota } from "../../../types";
 import {
   groupTransactionsByCategory,
   sortTransactions,
   TransactionsByCategoryMap,
   getLatestTransactionTime,
+  checkHasAppealProduct,
 } from "./NoQuotaCard";
 import { Cart } from "../../../hooks/useCart/useCart";
 import "../../../common/i18n/i18nMock";
 
 describe("NoQuotaCard utility functions", () => {
-  let sortedTransactions: PastTransactionsResult["pastTransactions"];
-  let allProducts: CampaignPolicy[];
-  let mockTransactionsByCategoryMap: TransactionsByCategoryMap;
+  const mockTransactionsByCategoryMap: TransactionsByCategoryMap = {
+    "TT token": {
+      transactions: [
+        {
+          header: "4 Aug 2020, 4:39PM",
+          details: "AAA987654321",
+          quantity: "1 qty",
+          isAppeal: false,
+          order: -1,
+        },
+      ],
+      hasLatestTransaction: true,
+      order: 1,
+    },
+    "Meal credits": {
+      transactions: [
+        {
+          header: "4 Aug 2020, 4:39PM",
+          details: "",
+          quantity: "$10",
+          isAppeal: false,
+          order: -1,
+        },
+        {
+          header: "4 Aug 2020, 4:38PM",
+          details: "",
+          quantity: "$5",
+          isAppeal: false,
+          order: -1,
+        },
+      ],
+      hasLatestTransaction: true,
+      order: 0,
+    },
+    "CDC Vouchers": {
+      transactions: [
+        {
+          header: "4 Aug 2020, 4:39PM",
+          details: "AAA987654322",
+          quantity: "1 book",
+          isAppeal: false,
+          order: -1,
+        },
+      ],
+      hasLatestTransaction: false,
+      order: 2,
+    },
+  };
 
   const latestTransactionTimeMs = 1596530350000;
   const latestTransactionTime = new Date(latestTransactionTimeMs);
 
-  beforeAll(() => {
-    sortedTransactions = [
-      {
-        category: "tt-token",
-        quantity: 1,
-        identifierInputs: [
-          {
-            ...defaultIdentifier,
-            label: "Device code",
-            value: "AAA987654321",
-          },
-        ],
-        transactionTime: latestTransactionTime,
-      },
+  describe("checkHasAppealProduct", () => {
+    const allQuotaResponseNoAppeal: Quota = {
+      globalQuota: [
+        {
+          category: "gift",
+          quantity: 0,
+          quotaRefreshTime: 1634800083079,
+          transactionTime: new Date("2020-10-21T07:08:03.079Z"),
```

Comment:
haha small nit, shall we use latestTransactionTime? 
```suggestion
          transactionTime: latestTransactionTime,
```